clusters/databricks_library.go

```diff
@@ -0,0 +1,103 @@
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/databrickslabs/terraform-provider-databricks/libraries"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceLibrary() *schema.Resource {
+	s := common.StructToSchema(libraries.Library{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		m["cluster_id"] = &schema.Schema{
+			Type:     schema.TypeString,
+			Required: true,
+		}
+		return m
+	})
+	return common.Resource{
+		Schema: s,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			clusterID := d.Get("cluster_id").(string)
+			err := NewClustersAPI(ctx, c).Start(clusterID)
+			if err != nil {
+				return err
+			}
+			var lib libraries.Library
+			err = common.DataToStructPointer(d, s, &lib)
+			if err != nil {
+				return err
+			}
+			librariesAPI := libraries.NewLibrariesAPI(ctx, c)
+			err = librariesAPI.Install(libraries.ClusterLibraryList{
+				ClusterID: clusterID,
+				Libraries: []libraries.Library{lib},
+			})
+			if err != nil {
+				return err
+			}
+			_, err = librariesAPI.WaitForLibrariesInstalled(libraries.Wait{
+				ClusterID: clusterID,
+				Timeout:   d.Timeout(schema.TimeoutCreate),
+				IsRunning: true,
+			})
+			if err != nil {
+				return err
+			}
+			d.SetId(fmt.Sprintf("%s/%s", clusterID, lib.String()))
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			split := strings.SplitN(d.Id(), "/", 2)
+			if len(split) != 2 {
+				split = []string{"unknown", "unknown"}
+			}
+			clusterID, libraryRep := split[0], split[1]
```

Comment:
move this into a separate function to avoid repetition? I think that was the same code in the cluster resource?